src/core/operations/RailFenceCipherDecode.mjs

```diff
@@ -0,0 +1,87 @@
+/**
+ * @author Flavio Diez [flaviofdiez+cyberchef@gmail.com]
+ * @copyright Crown Copyright 2020
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+
+/**
+ * Rail Fence Cipher Decode operation
+ */
+class RailFenceCipherDecode extends Operation {
+
+    /**
+     * RailFenceCipherDecode constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Rail Fence Cipher Decode";
+        this.module = "Ciphers";
+        this.description = "Decodes Strings that were created using the Rail fence Cipher provided a key and an offset";
+        this.infoURL = "https://en.wikipedia.org/wiki/Rail_fence_cipher";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Key",
+                type: "number",
+                value: 2
+            },
+            {
+                name: "Offset",
+                type: "number",
+                value: 0
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        const [key, offset] = args;
+
+        let cipher = input;
+
+        if (key < 2) {
+            return "Key has to be bigger than 2";
```

Comment:
Rather than returning errors as strings, I would recommend using the OperationError object. So this would become 'throw new OperationError("Key has to be bigger than 2");'

Reply:
good point. I will fix that ASAP... I think I saw somewhere similar being done like that and just went with it.